prereise/gather/demanddata/transportation_electrification/smart_charging.py

```diff
@@ -470,10 +490,26 @@ def smart_charging(
                 / (daily_vmt_total[day_iter] * 1000)
                 * emfacvmt
             )
+            model_year_profile[24:48] += (
+                outputelectricload[48:72]
+                / (daily_vmt_total[day_iter] * 1000)
+                * emfacvmt
+            )
 
-        else:
+        elif day_iter == len(input_day) - 2:
             # MW
             model_year_profile[day_iter * 24 : day_iter * 24 + 48] += (
+                outputelectricload[:48] / (daily_vmt_total[day_iter] * 1000) * emfacvmt
+            )
+            model_year_profile[:24] += (
+                outputelectricload[48:72]
+                / (daily_vmt_total[day_iter] * 1000)
+                * emfacvmt
+            )
+
+        else:
+            # MW
+            model_year_profile[day_iter * 24 : day_iter * 24 + 72] += (
```

Comment:
Not sure whether we can simplify the logic in a similar way above.

Reply:
Yeah, I’ve been thinking about what a generalized function for an arbitrary dwell window would look like, though it would likely be later…